Occasional panic in (*ChatSession).SendMessage(...) method

Version: `github.com/google/generative-ai-go v0.7.0`

The library may occasionally panic when calling the (*ChatSession).SendMessage(...) method:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x93a60d7]

goroutine 1 [running]:
github.com/google/generative-ai-go/genai.(*ChatSession).SendMessage(0xc0001e1880, {0x9cfe300, 0xa4613c0}, {0xc00088a070, 0x1, 0x1})
  /pkg/mod/github.com/google/generative-ai-go@v0.7.0/genai/chat.go:42 +0x197
```

After looking into the implementation, this can happen when `(m *GenerativeModel) generateContent(...)` returns the `iter.merged` value when the `err` variable equals `iterator.Done`, having this error returned on the first iteration.

A quick fix in `(m *GenerativeModel) generateContent(...)` can be:

```go
_, err := iter.Next()
if err == iterator.Done {
  if iter.merged == nil {
    return nil, err
  }
  return iter.merged, nil
}
```

If this suggestion is accepted, I can create a pull request with the fix.

The project using the SDK is [here](https://github.com/reugn/gemini-cli).
The full stack trace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x818c0d7]

goroutine 1 [running]:
github.com/google/generative-ai-go/genai.(*ChatSession).SendMessage(0xc000336bc0, {0x8ae4300, 0x92473c0}, {0xc000800100, 0x1, 0x1})
  /Users/reugn/go/pkg/mod/github.com/google/generative-ai-go@v0.7.0/genai/chat.go:42 +0x197
github.com/reugn/gemini-cli/gemini.(*ChatSession).SendMessage(...)
  /Users/reugn/github/reugn/gemini-cli/gemini/chat_session.go:32
github.com/reugn/gemini-cli/cli.(*geminiCommand).runBlocking(0xc0003d4180, {0xc000146240, 0x23})
  /Users/reugn/github/reugn/gemini-cli/cli/command.go:90 +0xb8
github.com/reugn/gemini-cli/cli.(*geminiCommand).run(0xc0003d4180, {0xc000146240, 0x23})
  /Users/reugn/github/reugn/gemini-cli/cli/command.go:82 +0x65
github.com/reugn/gemini-cli/cli.(*Chat).StartChat(0xc00009e040)
  /Users/reugn/github/reugn/gemini-cli/cli/chat.go:49 +0x51
main.run.func1(0xc0001ba400?, {0x869aae1?, 0x7?, 0x868fdfa?})
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:41 +0x9b
github.com/spf13/cobra.(*Command).execute(0xc00027a908, {0xc000036150, 0x0, 0x0})
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc00027a908)
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.run()
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:47 +0x114
main.main()
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:64 +0x13
```

@eliben, it took a while but reproduced with the latest main:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x6adc037]

goroutine 1 [running]:
github.com/google/generative-ai-go/genai.(*ChatSession).SendMessage(0xc000332c20, {0x74358d0, 0x7b9b3e0}, {0xc00024cdc0, 0x1, 0x1})
  /Users/reugn/go/pkg/mod/github.com/google/generative-ai-go@v0.8.1-0.20240301153401-90f89abc0d8d/genai/chat.go:42 +0x197
github.com/reugn/gemini-cli/gemini.(*ChatSession).SendMessage(...)
  /Users/reugn/github/reugn/gemini-cli/gemini/chat_session.go:32
github.com/reugn/gemini-cli/cli.(*geminiCommand).runBlocking(0xc00075c000, {0xc000873740, 0x33})
  /Users/reugn/github/reugn/gemini-cli/cli/command.go:99 +0xbb
github.com/reugn/gemini-cli/cli.(*geminiCommand).run(0xc00075c000, {0xc000873740, 0x33})
  /Users/reugn/github/reugn/gemini-cli/cli/command.go:91 +0x65
github.com/reugn/gemini-cli/cli.(*Chat).StartChat(0xc000168040)
  /Users/reugn/github/reugn/gemini-cli/cli/chat.go:55 +0x51
main.run.func1(0xc0001ae400?, {0x6feb4bc?, 0x7?, 0x6fe07e0?})
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:41 +0x9b
github.com/spf13/cobra.(*Command).execute(0xc00028a908, {0xc00011e120, 0x0, 0x0})
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:983 +0xaca
github.com/spf13/cobra.(*Command).ExecuteC(0xc00028a908)
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1115 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
  /Users/reugn/go/pkg/mod/github.com/spf13/cobra@v1.8.0/command.go:1039
main.run()
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:47 +0x1f5
main.main()
  /Users/reugn/github/reugn/gemini-cli/cmd/gemini/main.go:64 +0x13
```

--- a/genai/client.go
+++ b/genai/client.go
@@ -207,6 +207,9 @@ func (iter *GenerateContentResponseIterator) Next() (*GenerateContentResponse, e
 
 func protoToResponse(resp *pb.GenerateContentResponse) (*GenerateContentResponse, error) {
 	gcp := (GenerateContentResponse{}).fromProto(resp)
+	if gcp == nil {
+		return nil, errors.New("empty response from model")
+	}
 	// Assume a non-nil PromptFeedback is an error.
 	// TODO: confirm.
 	if gcp.PromptFeedback != nil && gcp.PromptFeedback.BlockReason != BlockReasonUnspecified {

